File: Pythonleetcode/Leetcode80.py
class Solution(object):
    def removeDuplicates(self, nums):
        if len(nums)<=2:
            return len(nums)
        fast=2
        slow=2
        while fast<len(nums):
            if nums[fast]!=nums[slow-2]:
                nums[slow] = nums[fast]
                slow+=1
            fast+=1
            # print(slow,fast,nums)
        return slow


print(Solution().removeDuplicates( nums = [1,1,1,2,2,3]))
print(Solution().removeDuplicates( nums = [0,0,1,1,1,1,2,3,3]))
print(Solution().removeDuplicates( nums = [1,1,1]))
print(Solution().removeDuplicates( nums = [1,1,1,1]))
print(Solution().removeDuplicates( nums = [1,2,2]))
print(Solution().removeDuplicates( nums = [1,2,2,2]))

File: CppPractice/CppLeetcode/Leetcode80.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) 
    {
        int slow = 0;
        unordered_map<int,int> theMap;
        for(int i=0;i<nums.size();i++)
        {
            nums[slow]=nums[i];
            if(theMap.find(nums[i])==theMap.end())
            {
                theMap[nums[i]]=1;
                slow++;
            }
            else if(theMap[nums[i]]==1)
            {
                theMap[nums[i]]+=1;
                slow++;
            }
        }
        return slow;
    }
};